src/test/java/org/owasp/validator/html/test/ESAPIInvalidPolicyTest.java

```diff
@@ -0,0 +1,76 @@
+package org.owasp.validator.html.test;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.InputStream;
+import java.lang.reflect.Method;
+
+import org.junit.After;
+import org.junit.Test;
+import org.owasp.validator.html.Policy;
+
+public class ESAPIInvalidPolicyTest {
+	/**
+	 * Property specified by the Antisamy project which may be used to disable
+	 * schema validation on policy files.
+	 */
+	private static final String ANTISAMY_PROJECT_PROP_SCHEMA_VALIDATION = "owasp.validator.validateschema";
+
+	@After
+	public void resetSystemProp() throws Exception {
+		System.clearProperty(ANTISAMY_PROJECT_PROP_SCHEMA_VALIDATION);
+		reloadSchemaValidation();
+	}
+
+	@Test
+	public void testDirectConfigAsBais() throws Exception {
+		Policy.setSchemaValidation(false);
+		InputStream stream = ESAPIInvalidPolicyTest.class.getResourceAsStream("/esapi-antisamy-InvalidPolicy.xml");
+		Policy.getInstance(toByteArrayStream(stream));
+	}
+
+	@Test
+	public void testDirectConfig() throws Exception {
+		Policy.setSchemaValidation(false);
+		InputStream stream = ESAPIInvalidPolicyTest.class.getResourceAsStream("/esapi-antisamy-InvalidPolicy.xml");
+		Policy.getInstance(stream);
+	}
+
+	@Test
+	public void testSystemProp() throws Exception {
+
+		System.setProperty(ANTISAMY_PROJECT_PROP_SCHEMA_VALIDATION, "false");
+		reloadSchemaValidation();
+		InputStream stream = ESAPIInvalidPolicyTest.class.getResourceAsStream("/esapi-antisamy-InvalidPolicy.xml");
+		Policy.getInstance(stream);
+	}
```

Comment:
Please invoke resetSystemProp() at the end of this test case so as to not affect other test cases that are expecting the default validation ON behavior.

Reply:
@davewichers - I think the `@After` annotation should take care of that.

Reply:
What does @After do? Not familiar with that. Does that push execution of the test case to the end after all others?

Reply:
It's a JUnit Annotation that causes the associated method to run **_after_** each of the test (i.e., those marked as `@Test`) completes. See https://junit.org/junit4/javadoc/4.12/org/junit/After.html for details. There is also a somewhat  corresponding `@Before` annotation. It is guaranteed to run, even if the test throws an exception. See Javadoc for details.

Reply:
Yes, it's done at the @After method. Even if the  test fails. It's a kind of glorified _finally_